firedrake/utility_meshes.py

```diff
@@ -934,6 +936,9 @@ def UnitIcosahedralSphereMesh(refinement_level=0, degree=1, reorder=None,
                                  degree=degree, reorder=reorder, comm=comm)
 
 
+# mesh is mainly used as a utility, so it's unnecessary to annotate the construction
+# in this case.
+@no_annotations
```

Comment:
There's a bunch more utility meshes that have interpolates in them, do they need the same?

Reply:
I added `no_annotations` to the `OctahedralSphereMesh` in particular because I was failing one test in `tests/firedrake_adjoint/test_shape_derivatives.py::test_PDE_hessian_neumann`. I wasn't able to debug what was causing the error and since annotating the construction of the mesh is not necessary, I decided to turn annotation off and it lead to the test passing. To allow me to make progress in my project, David and I decided we can use `no_annotations` as a temporary measure to allow me to move on. I can do the same to other utility meshes to make it more consistent if you want?